src/mobiflight.cpp

```diff
@@ -439,6 +455,43 @@ void ClearEncoders()
 #endif
 }
 
+#if MF_INPUT_SHIFTER_SUPPORT == 1
+//// INPUT SHIFT REGISTER /////
+void AddInputShifter(uint8_t latchPin, uint8_t clockPin, uint8_t dataPin, uint8_t modules, char const *name = "Shifter")
```

Comment:
**Issue warning**: the _name_ argument is not used anywhere in the function, therefore every shifter retains the same, default, name. All shifter chains will therefore send the same message for a given bit.
Currently, the "name" field of the object is initialized implicitly (to "InputShifter") during instantiation, and then it can no longer be modified; a method should be added to class MFInputShifter (e.g. `MFInputShifter::setName(const char *newName)` ) and invoked here.
Buttons behave differently because they are instantiated one by one, and thus named, upon registration.

Reply:
**Issue warning:** The _name_ argument is not used anywhere in the function, therefore every shifter retains the same, default, name. All shifter chains will therefore send the same message for a given bit.
Currently, the "name" field of the object is initialized implicitly (to "InputShifter") during instantiation, and then it can no longer be modified; a method should be added to class MFInputShifter (e.g. `MFInputShifter::setName(const char *newName) `) and invoked here.
Buttons behave differently because they are instantiated one by one, and thus named, upon registration.

Reply:
Good catch! This is fixed now, it just required assigning the name in the `attach` method, as is done with the other modules.

Reply:
Of course, that makes more sense than a separate method!